File: Python/variables.py
# variables are containers for storing data values
x = 5
y = 'John'
print(x, y)
# x = 'sally'
# print(x, y)

#Type casting
print(type(x))
print(type(y))

#variable names are case-sensitive
a = 4
A = 'sally'

'''
variable names are case-sensitive
variable names must start with a letter or underscore but not with a number
variable name cannot be a python keyword
'''

myVariableName = 'John' #--->camel case
MyVariableName  = 'John' #--->Pascal case
my_variable_name = 'John' # --->snake case

x, y, z, = 1, 2, 3
print(x, y, z)

x = y = z = 10
print(x, y, z)
print(x + y + z)

a = 'dont panic'
b = 42
# print(a + b) --->cannot combine a string with a number
print( a + str(b))

#Global variables--->can be used both inside and outside a function
x = 'awesome'

def myfunc():
    print('python is' + x)
myfunc()

File: OdinProject/js_basics/variables.js
// variables are containers for storing values
// var x = 5
// var y = 6

// console.log(x + y)

let x = 4
let y = 3
console.log(x + y)

x = 10
console.log(x + y)

// constant values cannot be changed

const price1 = 80
const price2 = 90
// price2 = 79
// console.log(price2)


// names may contain letters, digits , underscore and even '$'
// name must begin with either a letter or underscore or '$'
// names are case-sensitive
// JS keywords cannot be used as variable names

let person, car;

person = 'john'
car = "Volvo"
console.log(person, car)

// ES6 introduced const and let keyword which have Block scope
// Variables defined with let cannot be redeclared
// But using var keyword, we can redeclare variables again

let name = "ash"

/*
{  Not allowed since let has blockscope
    let a = 42
    let a = 89
}

{
let a = 42;
var a = 56
} 

{
    var a = 42;
    let a = 53
}

{
    let a = 42
}
// console.log(a) --->throws an error (let has block scope)
{
    var text = "hola"
}

console.log(text) 
// (var has global scope)

*/



// OPERATORS
/* ARITHMETIC OPERATORS( +, -, *, %, **, /)
   ASSIGNMENT OPERATORS( %=, +=, -=, *= , /=...)
   COMPARISION OPERATORS( ==, !+, >, <, >= , <=, ?)
   LOGICAL OPERATORS(&&, ||, !)

*/

let text1 = "John "
let text2 = "Doe"
console.log("Good Morning " + text1 + text2)

// DATA TYPES

// JS is a dynamically typed language 
// same variable can hold mulitple data types
let counter = 120
console.log(typeof counter)
counter = false
console.log(typeof counter)

let counter2; 
console.log(typeof counter2)


const cars = ['volvo', 'bmw', 'ferrari']
// we can access the array items using index
console.log(cars[0])

const person1 = {
    firstName : 'ash',
    lastName : 'Ketchum',
    age : 18
}


